src/backend/distributed/utils/citus_ruleutils.c

```diff
@@ -494,10 +542,18 @@ pg_get_tablecolumnoptionsdef_string(Oid tableRelationId)
 	relationName = generate_relation_name(tableRelationId, NIL);
 
 	relationKind = relation->rd_rel->relkind;
-	if (relationKind != RELKIND_RELATION && relationKind != RELKIND_FOREIGN_TABLE)
+	supportedRelationKind = (relationKind == RELKIND_RELATION || relationKind ==
+							 RELKIND_FOREIGN_TABLE);
+#if (PG_VERSION_NUM >= 100000)
+	supportedRelationKind = supportedRelationKind || relationKind ==
+							RELKIND_PARTITIONED_TABLE;
+#endif
+
+	if (!supportedRelationKind)
 	{
 		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
-						errmsg("%s is not a regular or foreign table", relationName)));
+						errmsg("%s is not a regular or foreign table or partitioned",
```

Comment:
Might make sense to error out a little earlier in `create_distributed_table`, but seems ok for now.